IoTuring/Entity/Deployments/Fanspeed/Fanspeed.py

```diff
@@ -0,0 +1,279 @@
+            self.SetEntitySensorValue(package.packageName, fans_above_threshold)
+            # Set extra attributes {fan name : fanspeed in rpm}
+            for i, fan in enumerate(package.attributes):
+                self.SetEntitySensorExtraAttribute(
+                    package.packageName,
+                    fan,
+                    readout[package.packageName][i].current,
+                    valueFormatterOptions=self.fanspeedFormatOptions,
+                )
+
+    @staticmethod
+    def above_threshold(fanspeeds: list[int], threshold: int) -> int:
+        """filters a list of integers for values above a threshold returns amount of values above threshold
+
+        :param fanspeeds: list of fanspeeds
+        :type fans: list[int]
+        :param threshold: threshold to filter values below
+        :type threshold: int
+        :return: amount of fanspeeds above threshold
+        :rtype: int
+        """
+        above_threshold_fans = [num for num in fanspeeds if num > threshold]
+        return len(above_threshold_fans)
+
+    @staticmethod
+    def parseControllernameFromInput(userInput) -> str:        
+        return list(psutil.sensors_fans().keys())[int(userInput)]
+
+
+    @staticmethod
+    def prettyprint_controllers(controllers) -> str:
+        """print available controllers
+
+        :param controllers: psutil.sensors_fans()
+        :type controllers: dict
+        :return: formated output as a string
+        :rtype: str
+        """
+        output = "\n"
+
+        for i, controller in enumerate(controllers.items()):
+            output += f"{i}: " + controller[0] + "\n"                          #controller \n
+            output += "\t"                                                  #   fan @ x rpm, 
+            for fan in controller[1]:
+                output += fan.label + " @ " + str(fan.current) + "rpm, "    #   fan @ x rpm, fan @ x rpm, fan @ x rpm, 
+            output += "\n"                                                  #\n 
+        return output
+
+    @classmethod
+    def ConfigurationPreset(cls) -> MenuPreset:
+        """generate the preset for human input, prints the names of available fancontrollers in the terminal
+
+        :return: preset 
+        :rtype: MenuPreset
+        """
+        preset = MenuPreset()
+        controllers = psutil.sensors_fans()
+        controllerNames = []
+        for name in controllers:
+            controllerNames.append(name)
+        preset.AddEntry(
+            f"which controller to watch\ncontrollers: {controllerNames}"
+            + Fanspeed.prettyprint_controllers(controllers),
+            CONFIG_KEY_CONTROLLER,
+            mandatory=True,
+            modify_value_callback=Fanspeed.parseControllernameFromInput
+        )
+        preset.AddEntry(
+            "At what threshold does a fan count as spinning",
+            CONFIG_KEY_THRESHOLD,
+            default=200,
+            mandatory=False
+        )
+        return preset
```

Comment:
The whole configuration/menupreset workflow is much simpler now, take a look how I rewrote this for disk entity, you can use a similar "select" configuration here.